src/atoms/Input/Input.tsx

```diff
@@ -0,0 +1,281 @@
+
+const InputWrapper: React.FC<InputWrapperProps> = ({
+  children,
+  inputWrapperStyle,
+}) => (
+  <div
+    style={{
+      position: "relative",
+      display: "flex",
+      justifyContent: "space-between",
+      alignItems: "center",
+      padding: "0 8px",
+      borderRadius: "4px",
+      ...inputWrapperStyle,
+    }}
+  >
+    {children}
+  </div>
+);
+
+const StyledIconWrapper = React.forwardRef<
+  HTMLDivElement,
+  StyledIconWrapperProps
+>(({ children }, ref) => (
+  <div
+    ref={ref}
+    style={{
+      backgroundColor: "#f0f0f0",
+      padding: "10.5px 12px",
+      borderRadius: "4px 0px 0px 4px",
+      display: "flex",
+      alignItems: "center",
+      height: "24px",
+      justifyContent: "center",
+    }}
+  >
+    {children}
+  </div>
+));
+
+const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  (
+    {
+      message,
+      variant = "default",
+      margin = "12px 0px",
+      align,
+      state,
+      value = "",
+      placeholder = "",
+      textMessage = "",
+      isTransparent = false,
+      Icon,
+      inputWrapperStyle,
+      useAnimatedPlaceholder = false,
+      disabled,
+      ...props
+    },
+    ref
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement | null>(null);
+    const iconWrapperRef = React.useRef<HTMLDivElement | null>(null);
+    const [iconWrapperWidth, setIconWrapperWidth] = React.useState(0);
+
+    React.useEffect(() => {
+      if (iconWrapperRef.current) {
+        setIconWrapperWidth(iconWrapperRef.current.offsetWidth);
+      }
+    }, [Icon, textMessage]);
+
+    const handleFocus = () => {
+      inputRef.current && inputRef.current.focus();
+    };
+
+    const ValidationIcon =
+      state === "success" ? "✔️" : state === "error" ? "❌" : null;
+
+    return (
+      <Wrapper margin={margin} align={align}>
+        <InputWrapper
+          inputWrapperStyle={{ ...inputWrapperStyle }}
+          disabled={disabled}
+        >
+          {(Icon || textMessage) && (
+            <StyledIconWrapper ref={iconWrapperRef}>
+              {Icon ? (
+                <Icon style={{ width: "24px", height: "24px" }} />
+              ) : (
+                <span>{textMessage}</span>
+              )}
+            </StyledIconWrapper>
+          )}
+          {useAnimatedPlaceholder && (
+            <AnimatedPlaceholder
+              placeholder={placeholder}
+              isInputFilled={!!value}
+              isIconPresent={!!Icon}
+              isTextMessage={!!textMessage}
+              onClick={handleFocus}
+              iconWrapperWidth={iconWrapperWidth}
+            />
+          )}
+          <input
+            ref={(current) => {
+              inputRef.current = current;
+              if (typeof ref === "function") ref(current);
+              else if (ref) ref.current = current;
+            }}
+            style={{
+              padding: "10px 12px",
+              width:
```

Comment:
create separate function which will return applicable width.
nesting ternary operators is not good.

Reply:
done